src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+    let nextState;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        nextState = {
+          ...currentState, ...extraData,
```

Comment:
Next state is being created without considering the default case in the switch statement. A default case should be added to handle unexpected action types.